shared/reducers/wallets.js

```diff
@@ -19,6 +19,47 @@ export default function(state: Types.State = initialState, action: WalletsGen.Ac
       return state.setIn(['paymentsMap', action.payload.accountID], I.List(action.payload.payments))
     case WalletsGen.selectAccount:
       return state.set('selectedAccount', action.payload.accountID)
+    case WalletsGen.validateAccountName:
+      return state.merge({accountNameError: '', accountNameValidationState: 'waiting'})
+    case WalletsGen.validatedAccountName:
+      return state.merge({
+        accountNameError: action.error ? action.payload.error : '',
+        accountNameValidationState: action.error ? 'error' : 'valid',
+      })
+    case WalletsGen.validateSecretKey:
+      return state.merge({secretKeyError: '', secretKeyValidationState: 'waiting'})
+    case WalletsGen.validatedSecretKey:
+      return state.merge({
+        secretKeyError: action.error ? action.payload.error : '',
+        secretKeyValidationState: action.error ? 'error' : 'valid',
+      })
+    case WalletsGen.clearErrors:
+      return state.merge({
+        accountNameError: '',
+        accountNameValidationState: 'none',
+        linkExistingAccountError: '',
+        secretKeyError: '',
+        secretKeyValidationState: 'none',
+      })
+    case WalletsGen.linkExistingAccount:
```

Comment:
could make this a fallthrough w/ clearErrors

Reply:
It ended up being a saga-only action